tests/unit/components/status/HeatMapColorBar.spec.js

```diff
@@ -0,0 +1,278 @@
+import { renderToString } from "@vue/server-test-utils";
+import { mount } from "@vue/test-utils";
+import { shallowMount } from "@vue/test-utils";
+import ComponentToTest from "@/components/status/HeatMapColorBar.vue";
+import { HeatMapColorBar as DistComponentToTest } from "@/dist/mantarray.common";
+
+import { createLocalVue } from "@vue/test-utils";
+
+let wrapper = null;
+
+const localVue = createLocalVue();
+
+describe("HeatMapColorBar.vue", () => {
+  // afterEach(() => wrapper.destroy());
+  test("When mounting HeatMapColorBar from the build dist file, Then it loads successfully and upper_range text '100 μN' is rendered as defined", () => {
+    const propsData = {
+      upper_range: "100",
+    };
+    wrapper = shallowMount(DistComponentToTest, {
+      propsData,
+      localVue,
+    });
+    const target_span = wrapper.find(".span__heatmap-scale-higher-value");
+    expect(target_span.text()).toStrictEqual("100 μN");
+  });
+  test("Given that the uuid, upper_range, lower_range and height, When mounted successfully, Then assert that upper_range,lower_range,height are applied and as uuid is match a default linear-gradient is applied on HeatMapColorBar", async () => {
+    const propsData = {
+      upper_range: "100",
+      lower_range: "0",
+      gradient_uuid: "0f81155d-23ec-4790-a3fe-92a6cc7c3c47",
+      heatmap_height: 481,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      localVue,
+    });
+    const target_span_higher = wrapper.find(
+      ".span__heatmap-scale-higher-value"
+    );
+    expect(target_span_higher.text()).toStrictEqual("100 μN");
+    const target_span_lower = wrapper.find(".span__heatmap-scale-lower-value");
+    expect(target_span_lower.text()).toStrictEqual("0 μN");
+
+    const target_div_gradient_holder = wrapper.find(
+      ".div__heatmap-gradient-holder"
+    );
+    expect(target_div_gradient_holder.attributes().style).toBe(
+      "height: 481px;"
+    );
+
+    const renderwrapper = await renderToString(ComponentToTest, {
+      propsData,
+      localVue,
+    });
+    expect(renderwrapper).toContain(
+      '<div data-server-rendered="true"><div class="div__heatmap-gradient-holder" style="height:481px;background:linear-gradient(to top,  #2c7bb6 0%,#00a6ca 12.5%,#00ccbc 25%,#90eb9d 37.5%,#ffff8c 50%,#f9d057 62.5%,#f29e2e 75%,#e76818 87.5%,#d7191c 100%);"></div> <span class="span__heatmap-scale-higher-value">100 μN</span> <span class="span__heatmap-scale-lower-value" style="top:465px;">0 μN</span></div>'
```

Comment:
testing the full contents of rendered HTML is a brittle way to test things. Sometimes it may be necessary, but it should be kept to a minimum.  Is there a reason you are checking that it contains this whole string? It seems like from the title of your test that you are really only interested in this section here `;background:linear-gradient(to top,  #2c7bb6 0%,#00a6ca 12.5%,#00ccbc 25%,#90eb9d 37.5%,#ffff8c 50%,#f9d057 62.5%,#f29e2e 75%,#e76818 87.5%,#d7191c 100%);"`

Could you reduce the amount of HTML being checked in this assertion so that the assertion wouldn't break just because (for example) we decided to refactor/rename this class name `span__heatmap-scale-higher-value`?

Please determine if this applies to your other tests as well

Reply:
Currently the vue-test-utils or JEST doesn't have any framework to test the modified properties such as `linear-gradient` in to assert if the values of the gradient is updated, this was only way the values can be asserted.

So as per your expectation changes cannot be done as there is no way we can only extract a part of CSS and assert those changes alone.

Just suggesting changes is not sufficient.

Please provide an alternative solution which works, I could not find any other way to validate so found out the way that was available and verified the same.

Reply:
I tagged you in a Pull Request demonstrating how to do this

Reply:
As the JEST and Vue-test-utils doesn't provide the essential asserting and reactive feature of the component. The question is for web artifacts, there are no testing required as per the general opinion and its a redundant testcase